src/entropic/sinkhorn.jl

```diff
@@ -209,3 +222,113 @@ function sinkhorn2(μ, ν, C, ε, alg::Sinkhorn; regularization=false, plan=noth
 
     return cost
 end
+
+function sinkhorn_loss(μ, ν, C, ε, alg::Sinkhorn; kwargs...)
+    return error(
+        "sinkhorn_loss is only implemented for alg::SinkhornGibbs. For other algorithms please use sinkhorn2",
+    )
+end
+
+"""
+    sinkhorn_divergence(μ::AbstractVecOrMat, ν::AbstractVecOrMat, C, ε, alg::Sinkhorn = SinkhornGibbs(), algμ = SymmetricSinkhornGibbs(), algν = SymmetricSinkhornGibbs(); regularization = nothing, plan = nothing, kwargs...)
+Compute the Sinkhorn Divergence between finite discrete
+measures `μ` and `ν` with respect to a common cost matrix `C`,
+entropic regularization parameter `ε` and algorithm `alg`. 
+
+For `regularization = true`, the Sinkhorn Divergence is that of [^FeydyP19] and is computed as:
+```math
+\\operatorname{S}_{ε}(μ,ν) := \\operatorname{OT}_{ε}(μ,ν)
+- \\frac{1}{2}(\\operatorname{OT}_{ε}(μ,μ) + \\operatorname{OT}_{ε}(ν,ν)),
+```
+where ``\\operatorname{OT}_{ε}(μ,ν)``, ``\\operatorname{OT}_{ε}(μ,μ)`` and
+``\\operatorname{OT}_{ε}(ν,ν)`` are the entropically regularized optimal transport loss 
+between `(μ,ν)`, `(μ,μ)` and `(ν,ν)`, respectively.
+
+For `regularization = false`, the Sinkhorn Divergence is that of [^GPC18] and is computed as:
+```math
+\\operatorname{S}_{ε}(μ,ν) := \\operatorname{W}_{ε}(μ,ν)
+- \\frac{1}{2}(\\operatorname{W}_{ε}(μ,μ) + \\operatorname{W}_{ε}(ν,ν)),
+```
+where ``\\operatorname{W}_{ε}(μ,ν)``, ``\\operatorname{W}_{ε}(μ,μ)`` and
+``\\operatorname{W}_{ε}(ν,ν)`` are the entropically regularized optimal transport *cost*
+between `(μ,ν)`, `(μ,μ)` and `(ν,ν)`, respectively, defined as 
+```math
+\\operatorname{W}_{ε}(μ, ν) = \\langle C, γ^\\star \\rangle
+```
+where ``γ^\\star`` is the entropy-regularised transport plan between `μ` and `ν`. 
+The default algorithm for computing the term ``\\operatorname{OT}_{ε}(μ, ν)`` is the `SinkhornGibbs` algorithm.
+For the terms ``\\operatorname{OT}_{ε}(μ, μ)`` and ``\\operatorname{OT}_{ε}(ν, ν)``, the symmetric fixed point iteration of [^FeydyP19] is used. 
+Alternatively, a pre-computed optimal transport `plan` between `μ` and `ν` may be provided.  
+
+[^GPC18]: Aude Genevay, Gabriel Peyré, Marco Cuturi, Learning Generative Models with Sinkhorn Divergences, Proceedings of the Twenty-First International Conference on Artficial Intelligence and Statistics, (AISTATS) 21, 2018
+[^FeydyP19]: Jean Feydy, Thibault Séjourné, François-Xavier Vialard, Shun-ichi Amari, Alain Trouvé, and Gabriel Peyré. Interpolating between optimal transport and mmd using sinkhorn divergences. In The 22nd International Conference on Artificial Intelligence and Statistics, pages 2681–2690. PMLR, 2019.
+See also: [`sinkhorn2`](@ref)
+"""
+function sinkhorn_divergence(
+    μ::AbstractVecOrMat,
+    ν::AbstractVecOrMat,
+    C,
+    ε,
+    alg::Sinkhorn=SinkhornGibbs(),
+    algμ::Sinkhorn=SymmetricSinkhornGibbs(),
+    algν::Sinkhorn=SymmetricSinkhornGibbs();
+    regularization=true,
+    plan=nothing,
+    kwargs...,
```

Comment:
Can we just pass along `kwargs...` and handle everything in the other functions? Possibly set `regularization = true` if we want to use it as default:
```suggestion
    regularization=true,
    kwargs...,
```